### PR TITLE
refactor(from): fromArray, fromPromise removed

### PR DESCRIPTION
**Pull request by kwonoj**
- removes fromArray and fromPromise to reduce API surface
- updates tests that utilized the above methods

covers issue https://github.com/ReactiveX/RxJS/issues/239 .

### CODE REVIEW

File: perf/micro/immediate-scheduler/observable/from-array.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = function fromArray(suite) {
     }
 
     var oldFromArrayWithImmediateScheduler = RxOld.Observable.fromArray(args, RxOld.Scheduler.immediate);
-    var newFromArrayWithImmediateScheduler = RxNew.Observable.fromArray(args);
+    var newFromArrayWithImmediateScheduler = RxNew.Observable.from(args);
```
Comment (kwonoj) [inline perf/micro/immediate-scheduler/observable/from-array.js]: Found this specific test case is now slower than old Rx's fromArray(). Same test for current thread scheduler still performs as expected. (New implementation is faster)

Comment (blesh) [inline perf/micro/immediate-scheduler/observable/from-array.js]: Interesting. I'll have a look, perhaps it's something we can optimize further.

Comment (kwonoj) [inline perf/micro/immediate-scheduler/observable/from-array.js]: More interestingly this morning gave another try on different machine, results was different saying new implementation is faster. Not sure if slower result came out by some of my system configuration. Anyway, will try couple of more run later.

Comment (blesh) [inline perf/micro/immediate-scheduler/observable/from-array.js]: I ran this a few times, and didn't at all see what you're seeing. RxJS Next was still faster, with no noticeable difference in speed from what's in master. Can you run again to confirm? I have serious doubts that making this function polymorphic and one assertion of `isArray` would make such a large difference.

I'm going to go ahead and merge for now. It's easy enough to add `fromArray` back in if we need to.

Comment (kwonoj) [inline perf/micro/immediate-scheduler/observable/from-array.js]: Will do, and create separate issue if I could find noticeable repro on my end.

Comment (blesh) [inline perf/micro/immediate-scheduler/observable/from-array.js]: Ah, I see, it's still faster than Rx2, but it's half the speed of just using fromArray. Amazing that one `isArray` check would make such a difference.

Comment (kwonoj) [inline perf/micro/immediate-scheduler/observable/from-array.js]: Thanks for clarification, it's obviously interesting. What would be best plan in this case?

Comment (blesh) [inline perf/micro/immediate-scheduler/observable/from-array.js]: Hmm... it looks like perhaps we should leave the other methods in the library, but guide people to use `from` unless they know for a fact their variable is an `Array`.

Again, I'm pretty shocked at the difference a single `Array.isArray` check made here. I've tried a variety of different methods, but I think we're going to have to walk away from this issue/PR

**Comment by blesh** [comment]
@kwonoj ... Thank you SO much for your effort on this, but after experimentation, it appears that the performance drop between `from(array)` and `fromArray(array)` makes this PR a no-go.

I'll have to close this, but we wouldn't have known this was the case if it wasn't for your efforts. As such, this was a very worthwhile PR.

File: perf/micro/immediate-scheduler/observable/from-array.js (hunk 1 above)
Comment (kwonoj) [inline perf/micro/immediate-scheduler/observable/from-array.js]: It seems makes sense. I'm also quite curious about why Array.isArray is taking those noticeable amount of time, still not sure though.

**Comment by kwonoj** [comment]
About to close PR on my end ;) Thanks.

**Pull request closed without merging**